src/v/raft/state_machine_manager.cc

```diff
@@ -444,6 +462,34 @@ void state_machine_manager::maybe_start_background_apply(
 ss::future<> state_machine_manager::background_apply_fiber(
   entry_ptr entry, ssx::semaphore_units units) {
     while (!_as.abort_requested() && entry->stm->next() < _next) {
+        if (entry->stm->next() < _raft->start_offset()) {
+            auto snapshot = co_await _raft->open_snapshot();
+            if (!snapshot) {
+                co_return;
```

Comment:
why do we bail if there is no snapshot? Should this be an error log at least?

Reply:
wow 😶‍🌫️